tendrl/gluster_integration/manager/manager.py

```diff
@@ -200,6 +266,27 @@ def on_pull(self, raw_data, cluster_id):
                     index += 1
                 except KeyError:
                     break
```

Comment:
Thats a good point. I kept the code as it was earlier. We can think of logging the error for the failed volume and continue to next one.